src/v3/stylis-logical-plugin/src/__tests__/emotion/emotion.tsx

```diff
@@ -0,0 +1,47 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import React from 'react';
+import createCache from '@emotion/cache';
+import { CacheProvider } from '@emotion/react';
+import { createSerializer } from '@emotion/jest';
+import styled from '@emotion/styled';
+import { render } from '@testing-library/preact';
+
+import logicalRtl from '../..';
+
+expect.addSnapshotSerializer(createSerializer());
+
+describe('with emotion environment', () => {
+  it('generates styles for simple scenario', async () => {
+    const body = document.body;
+    body.innerHTML = `
+      <div id="container" />
+    `;
+
+    const cache = createCache({
+      key: 'test',
+      stylisPlugins: [logicalRtl({ rootDirElement: '#container' })],
+      // @ts-expect-error
```

Comment:
what's the TS error?

Reply:
oh, this is because `document.querySelector` can return `Element | null` but `container` takes `Node | undefined` and i was trying to just get the test running. i'll fix so it doesn't have a TS error